packaging/test_rule_engine_plugin_metadata_guard.py

```diff
@@ -223,6 +223,49 @@ def test_plugin_honors_admin_only_config_option_when_rodsusers_manipulate_metada
         # Clean up.
         self.rods.assert_icommand(['imeta', 'rm', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
 
+    def test_plugin_disallows_rmw(self):
+        config = IrodsConfig()
+
+        # Set JSON configuration for the root collection.
+        root_coll = os.path.join('/', self.admin.zone_name)
+        json_config = json.dumps({
+            'prefixes': ['irods::'],
+            'admin_only': True
+        })
+        self.rods.assert_icommand(['imeta', 'set', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+        try:
+            with lib.file_backed_up(config.server_config_path):
+                self.enable_rule_engine_plugin(config)
+
+                # neither user should be able to use rmw
+                self.user.assert_icommand(['imeta', 'rmw', '-C', self.user.session_collection, '%', '%'], 'STDERR', ['SYS_NOT_ALLOWED'])
+                self.rods.assert_icommand(['imeta', 'rmw', '-C', self.rods.session_collection, '%', '%'], 'STDERR', ['SYS_NOT_ALLOWED'])
```

Comment:
Do we already cover rodsadmin level users too?

They shouldn't be allowed to use `rmw` either.

Reply:
i mean, this isn't about admin or not - this is the subcommand being disallowed completely.

so... i think this test is correct?   the rods user should get the SYS_NOT_SUPPORTED, as it shows above.

Reply:
I think so, I tried as irods on my testing container and it didn't let me rmw.
Based on where the check is in the code (before checking if admin), this checks out.

Reply:
True, but code can change and assuming anything about the implementation could lead to incorrect results.

Reply:
Correct

Reply:
okay.